### PR TITLE
feat: accept UUID in CUDA_VISIBLE_DEVICES round robin assignment

### PR DESCRIPTION
**Pull request by JoanFM**
**Goals:**
Accept UUID assignment when `ROUND-ROBIN` assignment of `CUDA_VISIBLE_DEVICES`.

### CODE REVIEW

File: docs/fundamentals/flow/topologies.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ You can restrict the visible devices in round-robin assignment using `CUDA_VISIB
 | 0          | 4          |
 
 
-You can restrict the visible devices in round-robin assignment by assigning a list of devices ids `CUDA_VISIBLE_DEVICES=RR1,3`. This creates the following assignment:
+You can restrict the visible devices in round-robin assignment by assigning a list of devices IDS `CUDA_VISIBLE_DEVICES=RR1,3`. This creates the following assignment:
```
Comment (alexcg1) [inline docs/fundamentals/flow/topologies.md:150]:
```suggestion
You can restrict the visible devices in round-robin assignment by assigning the list of device IDs to `CUDA_VISIBLE_DEVICES=RR1,3`. This creates the following assignment:
```

**Review by samsja — APPROVED** [review]
lgtm

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

File: jina/orchestrate/deployments/__init__.py [hunk 2]
```diff
@@ -742,13 +742,28 @@ def _parse_devices(value: str, num_devices: int):
...
+                # try to detect if parts are not numbers
+                try:
+                    int(parts[0])
+                except:
+                    use_uuids = True
+
+                if not use_uuids:
+                    return [int(p) for p in parts]
+                else:
+                    return parts
```
Comment (JohannesMessner) [inline jina/orchestrate/deployments/__init__.py:762]: Is it supported to have only some parts as UUID? Should we do this check for every part?

Comment (JoanFM) [reply jina/orchestrate/deployments/__init__.py:762]: I understand that if u pass the first as UUID, all of them will, other checks feel like overkill

**Pull request merged**